Napzak-iOS/Napzak-iOS/Network/Base/NetworkResult.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  NetworkResult.swift
+//  Napzak-iOS
+//
+//  Created by 조혜린 on 4/28/25.
+//
+
+import Foundation
+
+enum NetworkResult<T> {
+    case success(T?)                // 서버 통신 성공했을 때 (200번 대)
+    case badRequest                 // 올바르지 않은 요청 에러 발생했을 때 (400)
+    case unAuthorized               // 유효하지 않은 JWT 토큰으로 요청했을 때 (401)
+    case notFound                   // 존재하지 않는 api를 호출했을 때 (404)
+    case internalServerErr          // 서버 내부 오류가 발생했을 때 (500)
+    case networkFail                // 네트워크 연결 실패했을 때
+    case decodeErr                  // 데이터는 받아왔으나 DTO 형식으로 decode가 되지 않을 때
+}
```

Comment:
기존 NetworkResult는 에러를 구분해서 쓰긴 좋았지만,
실제 어떤 에러가 났는지 정확히 알기 어렵고,
코드도 복잡해질 수 있어서 유지보수가 힘들다고 느꼈어요

그래서 Result 타입에 맞게 별도의 Error 타입을 정의해야겠다고 판단했습니다. 
(혹여나 로컬라이징 경우가 생길까 봐 단순히 로컬라이징 가능한 Error로 했어요 그냥 Error해도 됩니다)
콜백 클로저 방식은 콜백 지옥이 생기기 쉬운 구조라, 

Result 타입을 통해 성공과 실패를 명확히 구분하고, 
더 깔끔하고 선언적인 흐름을 만드는 게 필요하다고 생각했습니다.

```suggestion
enum NetworkError: LocalizedError {
    case badRequest
    case unauthorized
    case notFound
    case internalServerError
    case decodingError
    case networkFail
    
    var errorDescription: String? {
        switch self {
        case .badRequest:
            return "잘못된 요청입니다"
        case .unauthorized:
            return "인증이 필요합니다"
        case .notFound:
            return "리소스를 찾을 수 없습니다"
        case .internalServerError:
            return "서버 오류가 발생했습니다"
        case .decodingError:
            return "데이터 변환 중 오류가 발생했습니다"
        case .networkFail:
            return "네트워크 연결에 실패했습니다"
        }
    }
}
```